src/app-gocardless/services/gocardless-service.js

```diff
@@ -438,6 +439,14 @@ export const goCardlessService = {
 
     handleGoCardlessError(response);
 
+    const bankAccount = BankFactory(institutionId);
+    response.transactions.booked = response.transactions.booked
+      .map(bankAccount.normalizeTransaction, true)
+      .filter((transaction) => transaction);
+    response.transactions.pending = response.transactions.pending
+      .map(bankAccount.normalizeTransaction, false)
+      .filter((transaction) => transaction);
```

Comment:
haha took me a few dozen of `console.log` to realise that's where the filtering logic was failing... 😅 
`booked` boolen was never passed to `normalizeTransaction` function and instead it was the transaction index from `map` function


```suggestion
    response.transactions.booked = response.transactions.booked
      .map((transaction) => bankAccount.normalizeTransaction(transaction, true))
      .filter((transaction) => transaction);
    response.transactions.pending = response.transactions.pending
      .map((transaction) =>
        bankAccount.normalizeTransaction(transaction, false),
      )
      .filter((transaction) => transaction);
```

Reply:
D'oh, that's so silly of me.  Good catch. :D